webui/src/Visualization.elm

```diff
@@ -1,64 +1,233 @@
+                            questionName
+
+                        Nothing ->
+                            ""
+                )
+                answers
+
+        resp =
+            Debug.log "List of questionNumbers" questions
     in
-    allResponses
+    questions
+
+
+getPoints : List AnswerResponse -> List Float
+getPoints answers =
+    let
+        points =
+            List.map
+                (\x ->
+                    toFloat x.points
+                )
+                answers
+
+        resp =
+            Debug.log "List of points" points
+    in
+    points
+
+
+getCategories : List AnswerResponse -> List String
+getCategories answers =
+    let
+        categories =
+            List.map
+                (\x ->
+                    x.category
+                )
+                answers
+
+        resp =
+            Debug.log "List of categories" categories
+    in
+    categories
+
+
+type alias AnswerResponse =
+    { answer : String
+    , category : String
+    , questionNumber : Int
+    , points : Int
+    }
+
+
+getAnswers : List Data.Survey.IpsativeQuestion -> List AnswerResponse
+getAnswers questions =
+    List.concatMap
+        (\x ->
+            let
+                answerList =
+                    List.map
+                        (\l ->
+                            let
+                                output =
+                                    { answer = l.answer
+                                    , category = l.category
+                                    , questionNumber = l.questionNumber
+                                    , points =
+                                        (Maybe.withDefault Data.Survey.emptyPointsAssigned
+                                            (List.head l.pointsAssigned)
+                                        ).points
+                                    }
+                            in
+                            output
+                        )
+                        x.answers
+
+                --   answer =
+                --       Maybe.withDefault Data.Survey.emptyAnswer
+                --          (List.Extra.find (\a -> "null" /= a.category) x.answers)
+                -- assigned =
+                --     Maybe.withDefault Data.Survey.emptyPointsAssigned
+                --        (List.head answer.pointsAssigned)
+            in
+            answerList
+        )
+        questions
 
 
-myVis : Spec
-myVis =
+type alias ColumnCount =
+    Maybe Int
+
+
+columnCount : ColumnCount
+columnCount =
+    Just 2
+
+
+spec : Data.Survey.IpsativeSurvey -> Spec
+spec model =
     let
-        _ =
+        responses =
             -- Do something here with some data eventually!
             getData
 
+        questions =
+            Zipper.toList model.questions
+
+        answerListing =
+            getAnswers questions
+
+        debugLog =
+            Debug.log "List of answers for questions" answerListing
+
+        -- TODO
```

Comment:
Are these comments still correct?

Reply:
I removed debugs and those old comments. Just now. Let me know what you think or if i missed anything.